packages/graphql/src/entities/Mutation.ts

```diff
@@ -119,71 +123,65 @@ export const mutation = mutationType({
-        const browsers = ctx.app.browserCache
+        if (!ctx.wizard.testingType) {
+          throw Error('Must set testingType before initializing a project')
+        }
+
+        // do not re-initialize plugins and dev-server.
+        if (ctx.wizard.testingType === 'component' && ctx.app.activeProject!.ctPluginsInitialized) {
+          debug('CT already initialized. Returning.')
+
+          return ctx.wizard
+        }
 
-        if (!browsers?.length) {
-          throw Error(`Need to call App#cacheBrowsers before opening a project.`)
+        if (ctx.wizard.testingType === 'e2e' && ctx.app.activeProject!.e2ePluginsInitialized) {
+          debug('E2E already initialized. Returning.')
+
+          return ctx.wizard
         }
 
+        /**
+         * Several things happen as part of initializing the open project.
+         * 1. Detect browsers (needed for plugins)
+         * 2. Run plugins (in cypress.config.js, setupNode or setupDevServer)
+         * 3. Open various servers, web sockets, etc.
+         */
+        debug('Detecting browsers...')
+        const browsers = await ctx.actions.getBrowsers()
+
+        debug('Found browsers: %o', browsers)
+        ctx.app.setBrowsers(browsers)
+
+        debug('initialize open_project for testingType %s', ctx.wizard.testingType)
         await ctx.actions.initializeOpenProject({
           ...ctx.launchArgs,
-          testingType: args.testingType,
-          config: {
-            browsers: browsers.map((x): BrowserContract => {
-              return {
-                name: x.name,
-                family: x.family,
-                majorVersion: x.majorVersion,
-                channel: x.channel,
-                displayName: x.displayName,
-                path: x.path,
-                version: x.version,
-              }
-            }),
-          },
-        }, ctx.launchOptions)
+          testingType: ctx.wizard.testingType,
+        }, ctx.launchOptions, browsers)
 
-        return ctx.app
+        debug('finishing initializing project')
+        ctx.wizard.navigate('forward')
+
+        return ctx.wizard
       },
     })
 
     t.field('launchOpenProject', {
       type: 'App',
       description: 'Launches project from open_project global singleton',
-      args: {
-        testingType: nonNull(TestingTypeEnum),
-      },
       async resolve (_root, args, ctx) {
-        if (!ctx.app.browserCache?.length) {
-          throw Error(`Need to call App#cacheBrowsers before opening a project.`)
-        }
-
-        const chrome = ctx.app.browserCache.find((x) => x.name === 'chrome')!
-
-        const browser: BrowserContract = {
-          name: chrome.name,
-          family: chrome.family,
-          majorVersion: chrome.majorVersion,
-          channel: chrome.channel,
-          displayName: chrome.displayName,
-          path: chrome.path,
-          version: chrome.version,
-        }
+        const browser = ctx.app.browsers.find((x) => x.name === 'chrome')!
```

Comment:
Another `!` that should go away

Reply:
Will do, was just trying to get it working - we need a lint rule, since it's quite easy to drop one of these and then forget it.